src/main/java/net/fabricmc/loader/game/MinecraftGameProvider.java

```diff
@@ -216,7 +216,9 @@ public boolean canOpenErrorGui() {
 	public void launch(ClassLoader loader) {
 		String targetClass = entrypoint;
 
-		if (envType == EnvType.CLIENT && targetClass.contains("Applet")) {
+		if (envType == EnvType.CLIENT && targetClass.contains("Runnable")) {
+			targetClass = "net.fabricmc.loader.entrypoint.rubydung.RubyDungLauncher";
```

Comment:
Wouldn't this crash, as `RubyDungLauncher` doesn't have a static `main(String[])` method to reflectively call?

Reply:
RubyDungLauncher does not have a main(String[] args), but has a constructor similar to AppletLauncher, and it doesn't crash? I don't understand what you mean.

Reply:
`AppletLauncher` isn't the class that gets reflected here, `AppletMain` is. When it's being used you end up with a chain of [`AppletMain#main`](https://github.com/FabricMC/fabric-loader/blob/62dec491951127559a597ea13bd0194ced70fe6a/src/main/java/net/fabricmc/loader/entrypoint/applet/AppletMain.java#L35) -> [`AppletFrame#launch`](https://github.com/FabricMC/fabric-loader/blob/62dec491951127559a597ea13bd0194ced70fe6a/src/main/java/net/fabricmc/loader/entrypoint/applet/AppletMain.java#L37) -> [`new AppletLauncher`](https://github.com/FabricMC/fabric-loader/blob/62dec491951127559a597ea13bd0194ced70fe6a/src/main/java/net/fabricmc/loader/entrypoint/applet/AppletFrame.java#L91-L96).

As it stands RubyDung won't take either route (as `targetClass` won't contain `Runnable` or `Applet`) so will be launched directly. `RubyDung#main` will spawn a thread to run the actual game from, leaving the main thread to complete (returning out from 228) and the JVM to wait for the game thread to stop (as it is not a daemon). If it did run through `RubyDungLauncher` (which I presume is what you're going for), you'd get a `NoSuchMethodException` from it not having a `main(String[])` method on line 227.

I think all you effectively need in `RubyDungLauncher` is this:
```java
public class RubyDungLauncher {
	public static void main(String[] args) {
		try {
			Runnable mc = FabricLauncherBase.getLauncher().getTargetClassLoader().loadClass("com.mojang.rubydung.RubyDung").asSubclass(Runnable.class).getDeclaredConstructor().newInstance();

			mc.run();
		} catch (ReflectiveOperationException e) {
			throw new RuntimeException("Error creating RubyDung instance", e);
		}
	}
}
```
But then the question arises whether this is even needed given the game can just be started directly, and if it does go wrong during `RubyDung#init` at least it calls `System.exit(0)` anyway.

Reply:
Uhh... It works? I don't understand what you mean when you say it should crash. Mind testing https://github.com/Cursed-Ancient-Fabric/preclassic-example-mod ? It works on my PC

Reply:
I'm saying if it actually used `RubyDungLauncher` it would crash. As explained above it's not actually doing so right now which is why you're not getting a crash testing

Reply:
I don't understand how RubyDungLauncher would be crashing? It just gets the entrypoint class, and then invokes the `run` method. But let me debug test it real quick.

Reply:
If RubyDungLauncher wasn't being ran, I don't understand why the game would open up as it is a Runnable class, and does not have a main method until rd-132328 (the next version)

Reply:
`RubyDungLauncher` wouldn't be the one crashing (although only from it not getting there, it would NPE on line 41 from `EntrypointTransformer.appletMainClass` being `null`), it will crash in this method on line 227 (as `c` will have no method called `main` taking a `String` array).

I downloaded rd-132211 for the purposes of doubling checking this PR and it definitely has a `main` method at the bottom of the class. Notch didn't have a `META-INF` to call it but it is there nonetheless.

Reply:
I don't have a main method in rd-132211 :concern:

Reply:
Alright, yeah this PR is borked.